### PR TITLE
Trigger e2e tests only for PRs that are ready for review

### PR DESCRIPTION
**Pull request by konstantin-msft**
- Trigger e2e tests only for PRs that are ready for review.

A follow-up for https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5649.

### CODE REVIEW

File: .github/workflows/msal-angular-e2e.yml [hunk 1]
```diff
@@ -28,7 +30,7 @@ permissions:
 
 jobs:
   run-e2e:
-    if: (github.repository == 'AzureAD/microsoft-authentication-library-for-js') && (github.actor != 'dependabot[bot]') && ((github.event.pull_request.head.repo.full_name == github.repository) || (github.event_name == 'push'))
+    if: (github.repository == 'AzureAD/microsoft-authentication-library-for-js') && (github.actor != 'dependabot[bot]') && ((github.event.pull_request.head.repo.full_name == github.repository) || (github.event_name == 'push')) && !github.event.pull_request.draft
```
Comment (tnorling) [inline .github/workflows/msal-angular-e2e.yml]: Big fan 👍

File: .github/workflows/msal-angular-e2e.yml [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
 
 on:
   push:
-    branches: 
+    branches:
       - dev
       - master
   pull_request:
-    paths: 
+    types: [opened, reopened, synchronize, ready_for_review]
```
Comment (tnorling) [inline .github/workflows/msal-angular-e2e.yml:9]: `synchronize` triggers on every commit so the rest become redundant here. I think what you're trying to do here is ready_for_review && any of the previous 3 triggers? The ready for review check would need to go below along with the draft check instead to achieve that.

I think I'd also be fine with only the draft check below for now if we can all be diligent about using draft PRs when still in progress.

Comment (konstantin-msft) [reply .github/workflows/msal-angular-e2e.yml:9]: We have to keep at least "synchronize" and "ready_for_review" to kick off e2e workflow once PR is ready for review without pushing new commits to the branch.

Comment (tnorling) [reply .github/workflows/msal-angular-e2e.yml:9]: Ah, my apologies, I misunderstood (was still thinking "approved"), thanks for clarifying.

**Review by tnorling — APPROVED** [review]

**Review by hectormmg — APPROVED** [review]

File: .github/workflows/msal-angular-e2e.yml (hunk 2 above)
Comment (sameerag) [reply .github/workflows/msal-angular-e2e.yml:9]: Can we have draft as the default setting when we open a PR for the repo?

**Review by sameerag — APPROVED** [review]
Minor comment on Draft PRs, lgtm otherwise.

File: .github/workflows/msal-angular-e2e.yml (hunk 2 above)
Comment (tnorling) [reply .github/workflows/msal-angular-e2e.yml:9]: Github defaults to whatever you personally used last. Don't think there is a repo wide setting.

**Review by peterzenz — APPROVED** [review]

**Pull request merged**